UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.IntegrationTests/RadioNavigationalWarnings/Admin/RadioNavigationalWarningEditAdminTest .cs

```diff
@@ -0,0 +1,150 @@
+            _fakeEditRadioNavigationalWarningsAdmin = GetFakeEditRadioNavigationalWarningsAdmin();
+            IActionResult result = await _controller.Edit(9, _fakeEditRadioNavigationalWarningsAdmin);
+            Assert.IsInstanceOf<IActionResult>(result);
+            Assert.AreEqual("Record updated successfully!", _controller.TempData["message"].ToString());
+            Assert.AreEqual("Index", ((RedirectToActionResult)result).ActionName);
+            
+        }
+
+
+        [Test]
+        public void WhenEditRadioNavigationWarningsWithInValidWarningType_ThenReturnInValidDataException()
+        {
+            _controller.TempData = _tempData;
+            _fakeEditRadioNavigationalWarningsAdmin = GetFakeEditRadioNavigationalWarningsAdmin();
+            _fakeEditRadioNavigationalWarningsAdmin.WarningType = 3;
+
+            Assert.ThrowsAsync(Is.TypeOf<InvalidDataException>(),
+                                async delegate { await _controller.Edit(_fakeEditRadioNavigationalWarningsAdmin.Id, _fakeEditRadioNavigationalWarningsAdmin); });
+        }
+
+
+        [Test]
+        public void WhenEditRadioNavigationWarningsWithInValidReference_ThenReturnArgumentNullException()
+        {
+            _controller.TempData = _tempData;
+            _fakeEditRadioNavigationalWarningsAdmin = GetFakeEditRadioNavigationalWarningsAdmin();
+            _fakeEditRadioNavigationalWarningsAdmin.Reference = string.Empty;
+
+            Assert.ThrowsAsync(Is.TypeOf<ArgumentNullException>(),
+                                async delegate { await _controller.Edit(_fakeEditRadioNavigationalWarningsAdmin.Id, _fakeEditRadioNavigationalWarningsAdmin); });
+        }
+
+        [Test]
+        public void WhenEditRadioNavigationWarningsWithInValidSummary_ThenReturnArgumentNullException()
+        {
+            _controller.TempData = _tempData;
+            _fakeEditRadioNavigationalWarningsAdmin = GetFakeEditRadioNavigationalWarningsAdmin();
+            _fakeEditRadioNavigationalWarningsAdmin.Summary = string.Empty;
+
+            Assert.ThrowsAsync(Is.TypeOf<ArgumentNullException>(),
+                                async delegate { await _controller.Edit(_fakeEditRadioNavigationalWarningsAdmin.Id, _fakeEditRadioNavigationalWarningsAdmin); });
+        }
+
+        [Test]
+        public void WhenEditRadioNavigationWarningsWithInValidContent_ThenReturnArgumentNullException()
+        {
+            _controller.TempData = _tempData;
+            _fakeEditRadioNavigationalWarningsAdmin = GetFakeEditRadioNavigationalWarningsAdmin();
+            _fakeEditRadioNavigationalWarningsAdmin.Content = string.Empty;
+
+            Assert.ThrowsAsync(Is.TypeOf<ArgumentNullException>(),
+                                async delegate { await _controller.Edit(_fakeEditRadioNavigationalWarningsAdmin.Id, _fakeEditRadioNavigationalWarningsAdmin); });
+        }
+
+        public static EditRadioNavigationalWarningsAdmin GetFakeEditRadioNavigationalWarningsAdmin()
```

Comment:
Local static methods should be declared as private.
```suggestion
        private static EditRadioNavigationalWarningsAdmin GetFakeEditRadioNavigationalWarningsAdmin()
```

Reply:
changes done